### PR TITLE
Bw/component modernization

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -** 
This PR modernizes the components within the GradesView component directory to hook components, and loads their redux data through new redux hooks.

Tech proposal doc: https://docs.google.com/document/d/19YwqYu_sRZZk_N4O_sfBU5OwJnEm8YDinfUzbwe-A78/edit

JIRA: [AU-1212](https://2u-internal.atlassian.net/browse/AU-1212)

**Developer Checklist**
- [x] Test suites passing
- [x] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

Smoke test of GradesView functionality

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Comment by nsprenkle** [comment]
Please resolve conflicts and update PR description to remove empty boilerplate before any walkthroughs happen.

**Comment by nsprenkle** [comment]
What would help me is to have a toy example in the description of what, "modernizes the components within the GradesView component directory to hook components, and loads their redux data through new redux hooks." looks like.

As someone who has not done these style of refactors, that is effectively useless information to me without an example 🙃
